tests/app/main/test_frameworks.py

```diff
@@ -4945,10 +4945,12 @@ def test_shows_updated_heading_and_countersigner_details_but_no_form_if_counters
         assert len(doc.xpath('//label[contains(text(), "I accept these proposed changes")]')) == 0
         assert len(doc.xpath('//input[@value="I accept"]')) == 0
 
-    def test_api_is_called_to_agree(self):
+    @mock.patch('app.main.views.frameworks.DMNotifyClient', autospec=True)
+    def test_api_is_called_to_agree(self, mocked_notify_class):
```

Comment:
I have no idea what this test name means 😹 I know it was there before but can we take this opportunity to make it more descriptive? Either that or combine it with `test_email_is_sent_to_correct_users` below (same scenario, each thing being asserted happens in both tests)